### PR TITLE
Add update sparse commitment

### PR DESCRIPTION
**Pull request by dragan2234**
Adding updateSparseCommitment with test. This can be merged after https://github.com/hyperledger/besu-native/pull/146

Closes https://github.com/hyperledger/besu-native/issues/129

### CODE REVIEW

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java [hunk 1]
```diff
@@ -75,4 +75,22 @@ public void testCallLibraryPedersenHash() {
         Bytes result = Bytes.of(LibIpaMultipoint.pedersenHash(total));
         assertThat(result).isEqualTo(Bytes32.fromHexString("0xff6e8f1877fd27f91772a4cec41d99d2f835d7320e929b8d509c5fa7ce095c51"));
     }
-}
+
+    @Test
+    public void testUpdateCommitmentSparse() {
+        // Numbers and result is taken from: https://github.com/crate-crypto/rust-verkle/blob/bb5af2f2fe9788d49d2896b9614a3125f8227818/ffi_interface/src/lib.rs#L576
+        // Identity element
+        byte[] old_commitment = new byte[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```
Comment (matkt) [inline ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java:83]: we are more using camelCase convention in this project. I think it will be better to keep the same convenstion everywhere in this code

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java [hunk 2]
```diff
@@ -75,4 +75,22 @@ public void testCallLibraryPedersenHash() {
         Bytes result = Bytes.of(LibIpaMultipoint.pedersenHash(total));
         assertThat(result).isEqualTo(Bytes32.fromHexString("0xff6e8f1877fd27f91772a4cec41d99d2f835d7320e929b8d509c5fa7ce095c51"));
     }
-}
+
+    @Test
+    public void testUpdateCommitmentSparse() {
```
Comment (matkt) [inline ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java:80]: https://github.com/crate-crypto/rust-verkle/blob/bb5af2f2fe9788d49d2896b9614a3125f8227818/ffi_interface/src/lib.rs#L529. maybe we need to add this test. it seems to add only test with commitment which is identity(empty) is just testing a part of the code

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java [hunk 3]
```diff
@@ -75,4 +75,22 @@ public void testCallLibraryPedersenHash() {
...
+        // Numbers and result is taken from: https://github.com/crate-crypto/rust-verkle/blob/bb5af2f2fe9788d49d2896b9614a3125f8227818/ffi_interface/src/lib.rs#L576
+        // Identity element
+        byte[] old_commitment = new byte[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+
+        byte[] old_scalar_new_scalar_index = new byte[]{2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```
Comment (matkt) [inline ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java:86]: why not having that in hex ? maybe simpler ?

**Comment by matkt** [comment]
build is failing because of rust version you need to update https://github.com/hyperledger/besu-native/blob/b3f626d1ae430850b948e99bf05ba31879226337/native-build.sh#L12 as we did for https://github.com/dragan2234/besu-native/blob/add-update-sparse-commitment/.github/workflows/build.yml#L30

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java (hunk 2 above)
Comment (dragan2234) [reply ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java]: @matkt thanks, added it now

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java (hunk 3 above)
Comment (dragan2234) [reply ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java]: @matkt fixed that now too

**Review by matkt — APPROVED** [review]
LGTM

**Comment by gfukushima** [comment]
Can we merge this?

**Comment by dragan2234** [comment]
@gfukushima From my side it's fine to merge, this is needed for the work on Java trie (it's not used currently).

**Review by matkt — APPROVED** [review]
LGTM
